pkg/event/target/webhook.go

```diff
@@ -208,6 +218,13 @@ func NewWebhookTarget(id string, args WebhookArgs, doneCh <-chan struct{}, logge
 
 	var store Store
 
+	if args.ClientCert != "" && args.ClientKey != "" {
+		transport.TLSClientConfig.GetClientCertificate = func(unused *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
Will use `tls.Config.Certificates` as per @aead 's suggestion.

To avoid calling `GetClientCertificate` multiple times for a single connection.

Reply:
GetClientCertificate should be implemented as a caching style, to reload the certs @Praveenrajmani @aead is not aware of the necessary requirements here.